simSPI/transfer.py

```diff
@@ -36,12 +41,14 @@ def ctf_freqs(n_pixels, psize=1.0, dim=2):
     """
     if dim == 1:
         freq_pix_1d = np.arange(0, 0.5, 1 / n_pixels)
-        freq_1d = freq_pix_1d * psize
+        freq_pix_1d_safe = freq_pix_1d[: n_pixels // 2]
+        freq_1d = freq_pix_1d_safe * psize
         return freq_1d
 
     # assert d == 2
     freq_pix_1d = np.arange(-0.5, 0.5, 1 / n_pixels)
-    x, y = np.meshgrid(freq_pix_1d, freq_pix_1d)
+    freq_pix_1d_safe = freq_pix_1d[:n_pixels]
```

Comment:
Good catch @geoffwoollard ! I'm still not 100% sure I follow but it seems to do the trick!